tests/DistIL.Tests/IR/MatchingTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿namespace DistIL.Tests.IR;
+
+using DistIL.AsmIO;
+using DistIL.IR;
+using DistIL.IR.Utils;
+
+[Collection("ModuleResolver")]
+public class MatchingTests
+{
+    private readonly ModuleResolver _modResolver;
+    private MethodDesc _stub;
+    private readonly ModuleDef _module;
+    private readonly TypeDef _testType;
+
+
+    public MatchingTests(ModuleResolverFixture mrf)
+    {
+        _modResolver = mrf.Resolver;
+        var type = _modResolver.Import(typeof(MatchingTests));
+        _stub = type.FindMethod("StubMethod");
+        _module = _modResolver.Create("Test");
+        _testType = _module.CreateType("Test", "Stub");
```

Comment:
This seems to be causing tests to fail when running them all at once (class is instantiated in parallel once per run(?), then gets a duplicate key exception).

I'm not sure how to best approach this but i think it would be easier to just pre-declare declare stub types/methods in the `tests/misc/TestAsm` project instead